lib/base/process.cpp

```diff
@@ -163,6 +132,13 @@ static Value ProcessSpawnImpl(struct msghdr *msgh, const Dictionary::Ptr& reques
 		(void)close(fds[1]);
 		(void)close(fds[2]);
 
+		for (auto& kv : extraEnv) {
+			if (setenv(kv.first.CStr(), kv.second.CStr(), 1) && errno != EINVAL) {
```

Comment:
Why do you want to ignore an `errno` of `EINVAL`?

According to [setenv(3) on Linux](https://man7.org/linux/man-pages/man3/setenv.3.html) (and POSIX, I guess), it will be returned if "name is NULL, points to a string of length 0, or contains an '=' character". I am not sure how portable this is, but at least [NetBSD's implementation](https://man.netbsd.org/setenv.3) also covers the value.

I would suggest checking if both the key and value of each `extraEnvironment` entry is not empty before emplacing them into `extraEnv` in the for loop above. Maybe also check for an `=` sign in the name up there. Afterwards, every `setenv` failure can be handled equally.

Reply:
But what for? setenv(3) checks it for me, so what? (https://docs.python.org/3/glossary.html#term-EAFP)

I mean, this whole PR is about not doing X if the stdlib can do it for me.

Reply:
My whole point was that `EINVAL` may differ on different platforms. If there are invalid `setenv` calls, let them fail and don't hide any error.